src/GoogleAuthenticator/RequestListener.php

```diff
@@ -32,15 +34,53 @@ class RequestListener
     protected $tokenStorage;
 
     /**
+     * NEXT_MAJOR: Remove this property
+     *
      * @var EngineInterface
      */
     protected $templating;
 
-    public function __construct(Helper $helper, TokenStorageInterface $tokenStorage, EngineInterface $templating)
+    /**
+     * @var Environment
+     */
+    private $twig;
+
+    /**
+     * NEXT_MAJOR: Remove `$templating` argument and make `$twig` argument mandatory
+     *
+     * @param EngineInterface|Environment $templating
+     */
+    public function __construct(Helper $helper, TokenStorageInterface $tokenStorage, object $templating = null, Environment $twig = null)
```

Comment:
I have also updated this class in order to use "twig" service if possible, respecting the BC promise. This is the only non final class which requires this logic.
I guess I'll send another PR to close the API with `@final` annotations covering this class too.